js/debug/job_recommendation.js

```diff
@@ -0,0 +1,53 @@
+(function ($) {
+    var click = function(jobId, recommender){
+        if(!JOBCENTRE.capabilities.localStorage)
+            return;
+
+        var clicks = {};
+        try {
+            clicks = JSON.parse(localStorage.getItem('recommendationInfo'));
+        } catch (e) {
+            // Continue if it fails
+        }
+
+        if(clicks == null)
+            clicks = {};
+
+        clicks[jobId] = {'recommender': recommender, 'timestamp': new Date().getTime()};
+        localStorage.setItem('recommendationInfo', JSON.stringify(clicks));
+    };
+
+    var view = function(jobId){
+        addEvent(jobId, "View");
+    };
+
+    var apply = function(jobId){
+        addEvent(jobId, "Apply");
+    };
+
+    var addEvent = function(jobId, jobEvent){
+        if(!JOBCENTRE.capabilities.localStorage || !localStorage.getItem('recommendationInfo'))
+            return;
+
+        var clicks;
+        try {
+            clicks = JSON.parse(localStorage.getItem('recommendationInfo'));
+        } catch (e) {
+            return;
+        }
+
+        var click = clicks[jobId];
+        // return if the jobEvent has already happened or if more than 24h have passed
+        if (!click || !click.timestamp || !click.recommender || click[jobEvent] || (new Date().getTime() - click.timestamp) / (1000 * 60 * 60) > 24)
+            return;
+
+        click[jobEvent] = true;
+        click.timestamp = new Date().getTime();
+
+        clicks[jobId] = click;
+        localStorage.setItem('recommendationInfo', JSON.stringify(clicks));
+        appInsights.trackEvent('JobseekerRecommendedJob' + jobEvent, {'jobId': jobId, 'recommender': click.recommender});
```

Comment:
Let's move all of this stuff up into try/catch.  That way, even if `click.timestamp` isn't a number, we won't get an exception and everything will fail gracefully.  We could then also simplify line 41 as we won't have to do so many checks.